### PR TITLE
key-rack: init at 0.4.0

### PR DESCRIPTION
**Pull request by getchoo**
## Description of changes

Key Rack allows to view and edit keys, like passwords or tokens, stored by apps. It is meant to eventually replace Seahorse (as explained [here](https://thisweek.gnome.org/posts/2024/06/twig-153/))

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ke/key-rack/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "sophie-h";
+    repo = "key-rack";
+    rev = "d6081b814ee2f91bc3d49734aeebcf7f512ac19a";
+    hash = "sha256-apAvethMUpsosKxk2DT310AIjsF755u36xm3cMYpRaM=";
+  };
+
+  postPatch = ''
+    patchShebangs {,.}*
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/key-rack/package.nix:29]:
```suggestion
    patchShebangs --build build-aux/{checks.sh,read-manifest.py}
```

**Review by Aleksanaa — COMMENTED** [review]
Please update to 0.4.0 (2 days ago). Others lgtm

File: pkgs/by-name/ke/key-rack/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  '';
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
```
Comment (Aleksanaa) [inline pkgs/by-name/ke/key-rack/package.nix:33]:
```suggestion
    inherit (finalAttrs) pname version src;
```

Just learned that, looks not bad

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
You need this:

https://github.com/NixOS/nixpkgs/blob/655a58a72a6601292512670343087c2d75d859c1/pkgs/by-name/pa/paper-plane/package.nix#L116-L123

**Comment by Aleksanaa** [comment]
(Our darwin builder is stuck?)

File: pkgs/by-name/ke/key-rack/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "View and edit your apps’ keys";
+    homepage = "https://gitlab.gnome.org/sophie-h/key-rack";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ getchoo ];
+    mainProgram = "key-rack";
```
[the remark this answers is not in the document]
Comment (getchoo) [reply pkgs/by-name/ke/key-rack/package.nix:67]: ...for once i thought one of these gnome apps might work on darwin :(
thanks though!

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/ke/key-rack/package.nix:67]: > ...for once i thought one of these gnome apps might work on darwin :(

Many work, but you'll only know after testing them

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**